src/deploy.ts

```diff
@@ -117,10 +124,15 @@ export async function deploy(gacFilename: string, deployConfig: DeployConfig) {
   return deploymentResult;
 }
 
-export async function deployProductionSite(gacFilename, projectId) {
+export async function deployProductionSite(
+  gacFilename,
+  productionDeployConfig: productionDeployConfig
+) {
+  const { projectId, target } = productionDeployConfig;
+
   const deploymentText = await execWithCredentials(
     "npx firebase-tools",
-    ["deploy", "--only", "hosting"],
+    ["deploy", !target ? "--only hosting" : "--only hosting:" + target],
```

Comment:
```suggestion
    ["deploy", "--only", `hosting${target ? ":" + target : ""}`],
```

Suggested change since we need `--only hosting` in both cases